lib/bundler/runtime.rb

```diff
@@ -127,16 +127,16 @@ def cache(custom_path = nil)
 
       Bundler.ui.info "Updating files in #{Bundler.settings.app_cache_path}"
 
-      # Do not try to cache specification for the gem described by the .gemspec
-      root_gem_name = nil
+      # Do not try to cache specification for the gem described any of the gemspecs
+      root_gem_names = nil
       if gemspec_cache_hash = Bundler.instance_variable_get(:@gemspec_cache)
-        gemspec = gemspec_cache_hash.values.first
-        root_gem_name = gemspec.name unless gemspec.nil?
+        gemspecs = gemspec_cache_hash.values
+        root_gem_names = gemspecs.map(&:name) unless gemspecs.empty?
```

Comment:
oh, `gemspec_cache_hash.values` will return an empty array if the hash is empty. Good call. Will remove the `unless`.